onnx_caffe2/backend.py

```diff
@@ -586,6 +588,45 @@ def prepare(cls, model, device='CPU', **kwargs):
         retval = Caffe2Rep(predict_net, ws, uninitialized)
         return retval
 
+    @classmethod
+    def _caff2_op_type(cls, onnx_op_type, opset_version):
+        broken_version = cls._broken_operators.get(onnx_op_type, float('Inf'))
+        if broken_version <= opset_version:
+            raise ValueError(
+                "Don't know how to translate op {} in ONNX operator set v{} \
+                (I only support prior to v{})".format(
+                    onnx_op_type, opset_version, broken_version))
+
+        c2_op_type = cls._renamed_operators.get(onnx_op_type, onnx_op_type)
+        if not core.IsOperator(c2_op_type):
+            raise ValueError(
+                "Don't know how to translate op {}".format(onnx_op_type))
+        return c2_op_type
+
+    @classmethod
+    def validate(cls, onnx_node, opset_version):
+        onnx_op_type = onnx_node.op_type
+        c2_op_type = cls._caff2_op_type(onnx_op_type, opset_version)
+        c2_schema = OpSchema.get(c2_op_type)
+
+        num_args = len(onnx_node.attrs)
+        if num_args != len(c2_schema.args):
+            raise ValueError(
+                "Number of arguments doesn't match for op: \
+                    {}".format(onnx_op_type))
+
+        num_inputs = len(onnx_node.inputs)
```

Comment:
We don't need to check num_inputs and num_outputs since they will be checked in caffe2.